apps/api/src/database/repositories/agent-repository.ts

```diff
@@ -78,6 +78,26 @@ export async function findByOwnerId(ownerId: string): Promise<SelectAgent[]> {
   }
 }
 
+/**
+ * Find agents by wallet address
+ * @param walletAddress Wallet address to search for
+ */
+export async function findByWalletAddress(
```

Comment:
(not sure, but i figure it'll be useful for staking/voting related stuff since those tables care about wallets > uuids)